### ISSUE TITLE
fails when filename would be too long

### ISSUE THREAD

**Issue by hominid** [flickr-downloadr/flickr-downloadr-gtk#25]
Any attempt to download a photos that would have a filename that's way too long on Win8.1 fails and gives the error "flickrDownloader has stopped working". It should fail more gracefully.

**Comment by hominid** [comment]
contents of Log Level All for crash:
<log4j:event logger="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" timestamp="1433520458044" level="DEBUG" thread="1"><log4j:message>ReadLoggingConfiguration</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" method="ReadLoggingConfiguration" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.MainClass" timestamp="1433520458065" level="INFO" thread="1"><log4j:message>Application Start.</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.MainClass" method="Main" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520458121" level="DEBUG" thread="1"><log4j:message>ctor</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method=".ctor" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520458169" level="DEBUG" thread="1"><log4j:message>AddTooltips</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="AddTooltips" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520458173" level="DEBUG" thread="1"><log4j:message>SetWelcomeLabel</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="SetWelcomeLabel" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520458177" level="DEBUG" thread="1"><log4j:message>AddSpinnerControl</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="AddSpinnerControl" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" timestamp="1433520458191" level="DEBUG" thread="1"><log4j:message>GetPresenter&lt;TView, TPresenter&gt;</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" method="GetPresenter" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520458345" level="DEBUG" thread="1"><log4j:message>ShowSpinner</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="ShowSpinner" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520458348" level="DEBUG" thread="1"><log4j:message>ShowLoggedOutControl</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="ShowLoggedOutControl" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520459363" level="DEBUG" thread="6"><log4j:message>ShowLoggedInControl</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="ShowLoggedInControl" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" timestamp="1433520459369" level="DEBUG" thread="6"><log4j:message>ReconfigureLogging</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" method="ReconfigureLogging" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520459371" level="DEBUG" thread="6"><log4j:message>SetWelcomeLabel</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="SetWelcomeLabel" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520459375" level="DEBUG" thread="6"><log4j:message>ShowSpinner</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="ShowSpinner" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520460239" level="DEBUG" thread="1"><log4j:message>buttonPrefsClick</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="buttonPrefsClick" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" timestamp="1433520460243" level="DEBUG" thread="1"><log4j:message>OpenPreferencesWindow</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.LoginWindow" method="OpenPreferencesWindow" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" timestamp="1433520460255" level="DEBUG" thread="1"><log4j:message>ctor</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" method=".ctor" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" timestamp="1433520460303" level="DEBUG" thread="1"><log4j:message>AddTooltips</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" method="AddTooltips" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" timestamp="1433520460308" level="DEBUG" thread="1"><log4j:message>setFieldsFromModel</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" method="setFieldsFromModel" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" timestamp="1433520460311" level="DEBUG" thread="1"><log4j:message>GetPresenter&lt;TView, TPresenter&gt;</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.Bootstrap.Bootstrapper" method="GetPresenter" file="" line="0" /></log4j:event>
<log4j:event logger="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" timestamp="1433520460316" level="DEBUG" thread="1"><log4j:message>SetCacheSize</log4j:message><log4j:properties><log4j:data name="log4jmachinename" value="white-tower" /><log4j:data name="log4japp" value="flickr-downloadr.exe" /><log4j:data name="log4net:Identity" value="" /><log4j:data name="log4net:UserName" value="WHITE-TOWER\Todd" /><log4j:data name="log4net:HostName" value="white-tower" /></log4j:properties><log4j:locationInfo class="FloydPink.Flickr.Downloadr.UI.Windows.PreferencesWindow" method="SetCacheSize" file="" line="0" /></log4j:event>

**Comment by mgroves** [comment]
Would you please provide example URL(s) so that I can try to repro this?

**Comment by hominid** [comment]
coming back to this after a bit... I pointed flickrDownloader at https://www.flickr.com/photos/kristenandtodd/ and asked it to download all of my photos and it crashed without giving any error messages after creating a directory of 81,044 images and .json files, from a projected 60,565 images & 60,565 .json = 121,130 files . When I tried a smaller set of just all public photos, it failed after downloading 20,042 files (of 28,035 projected images or a total of 56,070 files).

This is now on Windows 10, going into a directory at the root level with just one character, into the pretty verbose default folder hierarchy. I don't remember how I determined last time that the file name was the culprit, aside from trial and error.

I think I'm going to try running this in a Linux VM since the maximum file path length is much longer. I just want to get everything off of Flickr, once and for all.
